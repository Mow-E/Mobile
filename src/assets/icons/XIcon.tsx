import * as React from 'react';
import Svg, {SvgProps, Path} from 'react-native-svg';
import IconProps from './IconProps';
import colors from '../../styles/colors';

function SvgComponent(props: SvgProps): JSX.Element {
  return (
    <Svg width={24} height={24} viewBox="0 0 160 206" fill="none" {...props}>
      <Path
        stroke={props.stroke}
        strokeWidth={15}
        d="M0-7.5h241.8M0-7.5h241.8M0-7.5h241.8"
        transform="matrix(.5969 -.80231 .72118 .69274 11 205)"
      />
      <Path
        stroke={props.stroke}
        strokeWidth={15}
        d="M0-7.5h241.8M0-7.5h241.8M0-7.5h241.8"
        transform="matrix(.59663 .80251 -.87096 .49135 2 11)"
      />
    </Svg>
  );
}

/**
 * An icon for a check mark.
 */
function CheckMarkIcon({size = 24, colored = false}: IconProps): JSX.Element {
  const strokeColor =
    typeof colored === 'boolean' ? colors.gray['950'] : colored;

  return <SvgComponent height={size} width={size} stroke={strokeColor} />;
}

export default CheckMarkIcon;
