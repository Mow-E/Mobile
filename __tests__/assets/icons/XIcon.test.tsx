import 'react-native';
import React from 'react';
import XIcon from '../../../src/assets/icons/XIcon';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders default correctly', () => {
  const tree = renderer.create(<XIcon />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders colored correctly', () => {
  const tree = renderer.create(<XIcon colored />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders uncolored correctly', () => {
  const tree = renderer.create(<XIcon colored={false} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders in dark mode correctly', () => {
  const tree = renderer.create(<XIcon darkModeInverted />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders in different sizes', () => {
  const treeSmall = renderer.create(<XIcon size={24} />).toJSON();
  const treeLarge = renderer.create(<XIcon size={48} />).toJSON();

  expect(treeSmall).not.toEqual(treeLarge);
});
