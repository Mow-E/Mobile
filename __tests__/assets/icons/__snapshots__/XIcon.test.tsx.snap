// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders colored correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={206}
  vbWidth={160}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGPath
      d="M0-7.5h241.8M0-7.5h241.8M0-7.5h241.8"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      matrix={
        [
          0.5969,
          -0.80231,
          0.72118,
          0.69274,
          11,
          205,
        ]
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="15"
    />
    <RNSVGPath
      d="M0-7.5h241.8M0-7.5h241.8M0-7.5h241.8"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      matrix={
        [
          0.59663,
          0.80251,
          -0.87096,
          0.49135,
          2,
          11,
        ]
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="15"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders default correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={206}
  vbWidth={160}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGPath
      d="M0-7.5h241.8M0-7.5h241.8M0-7.5h241.8"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      matrix={
        [
          0.5969,
          -0.80231,
          0.72118,
          0.69274,
          11,
          205,
        ]
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="15"
    />
    <RNSVGPath
      d="M0-7.5h241.8M0-7.5h241.8M0-7.5h241.8"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      matrix={
        [
          0.59663,
          0.80251,
          -0.87096,
          0.49135,
          2,
          11,
        ]
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="15"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders in dark mode correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={206}
  vbWidth={160}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGPath
      d="M0-7.5h241.8M0-7.5h241.8M0-7.5h241.8"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      matrix={
        [
          0.5969,
          -0.80231,
          0.72118,
          0.69274,
          11,
          205,
        ]
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="15"
    />
    <RNSVGPath
      d="M0-7.5h241.8M0-7.5h241.8M0-7.5h241.8"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      matrix={
        [
          0.59663,
          0.80251,
          -0.87096,
          0.49135,
          2,
          11,
        ]
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="15"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders uncolored correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={206}
  vbWidth={160}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGPath
      d="M0-7.5h241.8M0-7.5h241.8M0-7.5h241.8"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      matrix={
        [
          0.5969,
          -0.80231,
          0.72118,
          0.69274,
          11,
          205,
        ]
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="15"
    />
    <RNSVGPath
      d="M0-7.5h241.8M0-7.5h241.8M0-7.5h241.8"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      matrix={
        [
          0.59663,
          0.80251,
          -0.87096,
          0.49135,
          2,
          11,
        ]
      }
      propList={
        [
          "stroke",
          "strokeWidth",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      strokeWidth="15"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
